spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -187,7 +191,7 @@ def env(self):
 
         # Environment variables that we need to pass to the kernel
         env_vars.update({
-            'SPY_EXTERNAL_INTERPRETER': not default_interpreter,
+            'SPY_EXTERNAL_INTERPRETER': self.path_to_custom_interpreter,
```

Comment:
I think this needs to be `not default_interpreter or self.path_to_custom_interpreter` (so when a custom interpreter is selected from the preferences the `SPY_EXTERNAL_INTERPRETER` variable is correctly set since the default interpreter is not being used):

```suggestion
            'SPY_EXTERNAL_INTERPRETER': not default_interpreter or self.path_to_custom_interpreter,
```